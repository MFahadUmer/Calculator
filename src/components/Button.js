import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value }) => (
  <button className={value === '=' ? 'equalButton' : 'button'} type="submit">
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
};
export default Button;
