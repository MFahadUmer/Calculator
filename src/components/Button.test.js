import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button Component Tests', () => {
  it('matches the button component snapshot', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
