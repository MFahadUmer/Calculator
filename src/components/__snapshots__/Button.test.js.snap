// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Tests matches the button component snapshot 1`] = `
<button
  className="width25 button"
  onClick=""
  style={
    Object {
      "backgroundColor": "orange",
    }
  }
  type="submit"
/>
`;
